C# warning: CS0618: 'TileMap' is obsolete

### Issue description

This is currently a minor annoyance, but every time I build my Godot C# project, I get these 4 warnings:

```
    [snip]/addons/phantom_camera/scripts/phantom_camera/PhantomCamera2D.cs(337,12): warning CS0618: 'TileMap' is obsolete: 'Use multiple 'TileMapLayer' nodes instead. To convert a TileMap to a set of TileMapLayer nodes,
    open the TileMap bottom panel with the node selected, click the toolbox icon in the top-right corner and choose 'Extract TileMap layers as individual TileMapLayer nodes'.' [[snip].csproj]
    [snip]/addons/phantom_camera/scripts/phantom_camera/PhantomCamera2D.cs(337,12): warning CS0618: 'TileMap' is obsolete: 'Use multiple 'TileMapLayer' nodes instead. To convert a TileMap to a set of TileMapLayer nodes,
    open the TileMap bottom panel with the node selected, click the toolbox icon in the top-right corner and choose 'Extract TileMap layers as individual TileMapLayer nodes'.' [[snip].csproj]
    [snip]/addons/phantom_camera/scripts/phantom_camera/PhantomCamera2D.cs(235,32): warning CS0618: 'TileMap' is obsolete: 'Use multiple 'TileMapLayer' nodes instead. To convert a TileMap to a set of TileMapLayer nodes,
    open the TileMap bottom panel with the node selected, click the toolbox icon in the top-right corner and choose 'Extract TileMap layers as individual TileMapLayer nodes'.' [[snip].csproj]
    [snip]/addons/phantom_camera/scripts/phantom_camera/PhantomCamera2D.cs(339,29): warning CS0618: 'TileMap' is obsolete: 'Use multiple 'TileMapLayer' nodes instead. To convert a TileMap to a set of TileMapLayer nodes,
    open the TileMap bottom panel with the node selected, click the toolbox icon in the top-right corner and choose 'Extract TileMap layers as individual TileMapLayer nodes'.' [[snip].csproj]
```

[TileMapLayer](https://docs.godotengine.org/en/4.3/classes/class_tilemaplayer.html) was [introduced](https://godotengine.org/releases/4.3/#highlights-new-tilemaplayer-node) in Godot 4.3 (August '24) and deprecated TileMap at the same time.

The class is PhantomCamera2D, and the lines on `main` have diverged a bit from 0.9.3.1, but the TileMap classes are still there.

Personally I'd be fine if there was some way/annotation/pragma to suppress these warnings without removing TileMap, but it depends on which versions of Godot this library aims to support. In the meantime, adding `<NoWarn>CS0618</NoWarn>` to the .csproj suppresses this warning, but for the whole project (including my code), which isn't ideal.

### Steps to reproduce

I believe any project using Godot 4.5 with C# and this addon will see this warning when building on the command line.

### (Optional) Minimal reproduction project

_No response_

--- a/addons/phantom_camera/scripts/phantom_camera/PhantomCamera2D.cs
+++ b/addons/phantom_camera/scripts/phantom_camera/PhantomCamera2D.cs
@@ -218,11 +218,6 @@ public class PhantomCamera2D : PhantomCamera
         Node2D.Connect(SignalName.NoiseEmitted, callableNoiseEmitted);
     }
 
-    public void SetLimitTarget(TileMap tileMap)
-    {
-        Node2D.Call(MethodName.SetLimitTarget, tileMap.GetPath());
-    }
-
     public void SetLimitTarget(TileMapLayer tileMapLayer)
     {
         Node2D.Call(MethodName.SetLimitTarget, tileMapLayer.GetPath());
@@ -314,16 +309,10 @@ public class PhantomCamera2D : PhantomCamera
 
 public class LimitTargetQueryResult(GodotObject godotObject)
 {
-    public bool IsTileMap => godotObject.IsClass("TileMap");
-
     public bool IsTileMapLayer => godotObject.IsClass("TileMapLayer");
 
     public bool IsCollisionShape2D => godotObject.IsClass("CollisionShape2D");
 
-    public TileMap? AsTileMap()
-    {
-        return IsTileMap ? (TileMap)godotObject : null;
-    }
 
     public TileMapLayer? AsTileMapLayer()
     {

--- a/addons/phantom_camera/scripts/phantom_camera/phantom_camera_2d.gd
+++ b/addons/phantom_camera/scripts/phantom_camera/phantom_camera_2d.gd
@@ -410,22 +410,22 @@ var _should_rotate_with_target: bool = false
 	set = set_limit_bottom,
 	get = get_limit_bottom
 
-## Allows for setting either a [TileMap], [TileMapLayer] or [CollisionShape2D] node to
+## Allows for setting either a [TileMapLayer] or [CollisionShape2D] node to
 ## automatically apply a limit size instead of manually adjusting the Left,
 ## Top, Right and Left properties.[br][br]
-## [b]TileMap / TileMapLayer[/b][br]
-## The Limit will update after the [TileSet] of the [TileMap] / [TileMapLayer] has changed.[br]
+## [b]TileMapLayer[/b][br]
+## The Limit will update after the [TileSet] of the [TileMapLayer] has changed.[br]
 ## [b]Note:[/b] The limit size will only update after closing the TileMap editor
 ## bottom panel.
 ## [br][br]
 ## [b]CollisionShape2D[/b][br]
 ## The limit will update in realtime as the Shape2D changes its size.
 ## Note: For performance reasons, resizing the [Shape2D] during runtime will not change the Limits sides.
-@export_node_path("TileMap", "Node2D", "CollisionShape2D") var limit_target: NodePath = NodePath(""):
+@export_node_path("TileMapLayer", "CollisionShape2D") var limit_target: NodePath = NodePath(""):
 	set = set_limit_target,
 	get = get_limit_target
 
-## Applies an offset to the [TileMap]/[TileMapLayer] Limit or [Shape2D] Limit.
+## Applies an offset to the [TileMapLayer] Limit or [Shape2D] Limit.
 ## The values goes from [param Left], [param Top], [param Right]
 ## and [param Bottom].
 @export var limit_margin: Vector4i = Vector4.ZERO:
@@ -1053,8 +1053,8 @@ func update_limit_all_sides() -> void:
 		_limit_sides.y = limit_top
 		_limit_sides.z = limit_right
 		_limit_sides.w = limit_bottom
-	elif _limit_node is TileMap or _limit_node.is_class("TileMapLayer"):
-		var tile_map := _limit_node
+	elif _limit_node is TileMapLayer:
+		var tile_map: TileMapLayer = _limit_node
 
 		if not tile_map.tile_set: return # TODO: This should be removed once https://github.com/godotengine/godot/issues/96898 is resolved
 
@@ -1604,17 +1604,17 @@ func set_limit_target(value: NodePath) -> void:
 	# Waits for PCam2d's _ready() before trying to validate limit_node_path
 	if not is_node_ready(): await ready
 
-	# Removes signal from existing TileMap node
+	# Removes signal from existing TileMapLayer node
 	if is_instance_valid(get_node_or_null(value)):
 		var prev_limit_node: Node2D = _limit_node
 		var new_limit_node: Node2D = get_node(value)
 
 		if prev_limit_node:
-			if prev_limit_node is TileMap or prev_limit_node.is_class("TileMapLayer"):
+			if prev_limit_node is TileMapLayer:
 				if prev_limit_node.changed.is_connected(_on_tile_map_changed):
 					prev_limit_node.changed.disconnect(_on_tile_map_changed)
 
-		if new_limit_node is TileMap or new_limit_node.is_class("TileMapLayer"):
+		if new_limit_node is TileMapLayer:
 			if not new_limit_node.changed.is_connected(_on_tile_map_changed):
 				new_limit_node.changed.connect(_on_tile_map_changed)
 		elif new_limit_node is CollisionShape2D:
@@ -1626,7 +1626,7 @@ func set_limit_target(value: NodePath) -> void:
 				limit_target = ""
 				return
 		else:
-			printerr("Limit Target is not a TileMap, TileMapLayer or CollisionShape2D node")
+			printerr("Limit Target is not a TileMapLayer or CollisionShape2D node")
 			return
 	elif value == NodePath(""):
 		reset_limit()

